python/lsst/analysis/tools/tasks/catalogMatch.py

```diff
@@ -0,0 +1,408 @@
+    )
+
+    matchCatalog = pipeBase.connectionTypes.Output(
+        doc="Catalog with matched target and reference objects with separations",
+        name="{targetCatalog}_{refCatalog}_match",
+        storageClass="DataFrame",
+        dimensions=("tract", "skymap")
+    )
+
+
+class CatalogMatchConfig(pipeBase.PipelineTaskConfig, pipelineConnections=CatalogMatchConnections):
+
+    astrometryRefObjLoader = pexConfig.ConfigurableField(
+        target=LoadReferenceObjectsTask,
+        doc="Reference object loader for astrometric fit",
+    )
+
+    matcher = pexConfig.ConfigurableField(
+        target=AstropyMatchTask,
+        doc="Task for matching refCat and SourceCatalog"
+    )
+
+    epoch = pexConfig.Field(
+        doc="Epoch to which reference objects are shifted",
+        dtype=float,
+        default=2015.0
+    )
+
+    selectorBand = pexConfig.Field(
+        doc="Band to use when selecting objects, primarily for extendedness",
+        dtype=str,
+        default='i'
+    )
+
+    selectorActions = ConfigurableActionStructField(
+        doc="Which selectors to use to narrow down the data for QA plotting.",
+        default={"flagSelector": CoaddPlotFlagSelector},
+    )
+
+    sourceSelectorActions = ConfigurableActionStructField(
+        doc="What types of sources to use.",
+        default={"sourceSelector": StellarSelector},
+    )
+
+    extraColumnSelectors = ConfigurableActionStructField(
+        doc="Other selectors that are not used in this task, but whose columns"
+            "may be needed downstream",
+        default={"selector1": SnSelector,
+                 "selector2": GalacticSelector}
+    )
+
+    extraColumns = pexConfig.ListField(
+        doc="Other catalog columns to persist to downstream tasks",
+        dtype=str,
+        default=['i_cModelFlux', 'x', 'y']
+    )
+
+    def setDefaults(self):
+        self.astrometryRefObjLoader.requireProperMotion = False
+        self.astrometryRefObjLoader.anyFilterMapsToThis = 'phot_g_mean'
+
+
+class CatalogMatchTask(pipeBase.PipelineTask):
+    """Match a tract-level catalog to a reference catalog
+    """
+
+    ConfigClass = CatalogMatchConfig
+    _DefaultName = "catalogMatch"
+
+    def __init__(self, butler=None, initInputs=None, **kwargs):
+        super().__init__(**kwargs)
+        self.makeSubtask("matcher")
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        # Docs inherited from base class
+
+        inputs = butlerQC.get(inputRefs)
+
+        columns = ['coord_ra', 'coord_dec', 'patch'] + self.config.extraColumns.list()
+        for selectorAction in [self.config.selectorActions, self.config.sourceSelectorActions,
+                               self.config.extraColumnSelectors]:
+            for selector in selectorAction:
+                for band in ["g", "r", "i", "z", "y"]:
+                    selectorSchema = selector.getFormattedInputSchema(band=band)
+                    columns += [s[0] for s in selectorSchema]
```

Comment:
Mixture of ' and ". I don't think it matters but it looks neater when it is consistent.

Reply:
This was moved to a config option (with consistent quotation marks).